### PR TITLE
Added workflow to build project on different compilers

### PR DESCRIPTION
**Pull request by codeinred**
Added workflow to build project on different compilers

### CODE REVIEW

**Comment by codeinred** [comment]
NB: the build fails on windows, but this isn't my fault. This branch was directly derived from main, and the code that's failing is the code before I made any changes or modifications

File: .github/workflows/cmake.yml [hunk 1]
```diff
@@ -0,0 +1,134 @@
+name: CMake
+
+on: [push]
+
+env:
+  # Customize the CMake build type here (Release, Debug, RelWithDebInfo, etc.)
+  BUILD_TYPE: Release
```
Comment (s5bug) [inline .github/workflows/cmake.yml]: Please change this to `Debug` for push events, and `RelWithDebInfo` for release/manual/pre-merge events (or alternatively don't build on `Debug` at all and only build on `RelWithDebInfo`)

**Review by s5bug — CHANGES_REQUESTED** [review]

**Comment by s5bug** [comment]
> NB: the build fails on windows, but this isn't my fault. This branch was directly derived from main, and the code that's failing is the code before I made any changes or modifications

Yep, I can reproduce this on my own machine. I don't know how I didn't catch this considering my main dev machine is Windows...

**Comment by codeinred** [comment]
I switched from Release to RelWithDebugInfo!

**Review by s5bug — APPROVED** [review]
Looks good! When I'm free next (probably Thursday?) I'll run a PR by you for a few of my personal changes.

**Pull request merged**